src/tools/perf/lib/libperf_int.h

```diff
@@ -34,14 +34,28 @@ typedef struct ucp_perf_request  ucp_perf_request_t;
 
 
 struct ucx_perf_allocator {
+    ucs_memory_type_t mem_type;
     ucs_status_t (*init)(ucx_perf_context_t *perf);
-    ucs_status_t (*ucp_alloc)(ucx_perf_context_t *perf, size_t length,
+    ucs_status_t (*ucp_alloc)(const ucx_perf_context_t *perf, size_t length,
                               void **address_p, ucp_mem_h *memh, int non_blk_flag);
-    void         (*ucp_free)(ucx_perf_context_t *perf, void *address,
+    void         (*ucp_free)(const ucx_perf_context_t *perf, void *address,
                              ucp_mem_h memh);
-    void*        (*memset)(void *s, int c, size_t len);
+    ucs_status_t (*uct_alloc)(const ucx_perf_context_t *perf, size_t length,
+                              unsigned flags, void **address_p,
+                              uct_mem_h *memh_p, void **ctx_p);
+    void         (*uct_free)(const ucx_perf_context_t *perf, void *address,
+                             uct_mem_h memh, void *ctx);
+    void         (*memcpy)(void *dst, ucs_memory_type_t dst_mem_type,
+                           const void *src, ucs_memory_type_t src_mem_type,
+                           size_t count);
+    void*        (*memset)(void *dst, int value, size_t count);
 };
 
+typedef struct ucx_perf_uct_mem {
+    void                         *ctx;
+    uct_mem_h                    memh;
```

Comment:
it is already a part of `uct_allocated_memory_t`

Reply:
fixed